### PR TITLE
Adds SetCodecPreferences to RTPTransceiver

### PR DESCRIPTION
**Pull request by digitalix**
#### Description

Adds option to set chosen codecs on individual Transceiver.

### CODE REVIEW

File: rtptransceiver.go [hunk 1]
```diff
@@ -16,6 +17,10 @@ type RTPTransceiver struct {
 	receiver  atomic.Value // *RTPReceiver
 	direction atomic.Value // RTPTransceiverDirection
 
+	me     *MediaEngine
```
Comment (Sean-Der) [inline rtptransceiver.go:20]: Can you embed `api *API` instead? That is just the common pattern across `RTC*` objects we have right now.

File: rtptransceiver.go [hunk 2]
```diff
@@ -106,6 +185,11 @@ func (t *RTPTransceiver) Stop() error {
 }
 
 func (t *RTPTransceiver) setReceiver(r *RTPReceiver) {
+	if r != nil {
+		t.mu.RLock()
+		r.codecs = t.codecs
```
Comment (Sean-Der) [inline rtptransceiver.go:190]: Is it possible that these could get out of date? Maybe we should store the `RTPTransceiver` on the `RTPReceiver` and  `RTPSender` as well.

Comment (digitalix) [reply rtptransceiver.go]: You're right, it's better if we store the `RTPTransceiver ` on the `RTPReceiver` and `RTPSender`. Done.

**Comment by Sean-Der** [comment]
Moved to PR to branch I have push access on https://github.com/pion/webrtc/pull/1848

**Pull request closed without merging**